src/installer/installer.go

```diff
@@ -304,8 +304,16 @@ func (i *installer) waitForController() error {
 		return err
 	}
 	predicate := func() bool {
-		return common.IsPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
+		controllerPod := common.GetPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
 			map[string]string{"job-name": assistedControllerPrefix}, v1.PodRunning, i.log)
+		if controllerPod != nil {
+			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
+			// if failed to upload logs, log why and continue
+			if err != nil {
+				i.log.WithError(err).Warnf("Failed to upload controller logs")
```

Comment:
Note that in case of failure it will not continue in the loop but return true in the next line

Reply:
yap, done on purpose. Sort of best effort. Don't want to break installation on this case